#!/usr/bin/env python

PACKAGE = "m100_trajectory"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       	Name			Type      	Level	Description				Default 	Min	Max

gen.add(	"traj_on",		bool_t,		0,	"Trajectory on flag", 			False)
gen.add(	"adaptive_yaw_on",	bool_t,		0,	"Adaptive yaw on flag", 		False)
gen.add(	"reg_on",		bool_t,		0,	"Regression on flag",			False)
gen.add(	"land",			bool_t,		0,	"Land to ground", 			False)
gen.add("pub_on_setpoint_position",	bool_t,		0,	"Publish flag on /mavros/setpoint_position/local", False)

gen.add( 	"yaw_hover",		int_t, 		0, 	"Hover yaw angle [deg]",  		0, 	-180, 	180)
gen.add( 	"wall_dist",		double_t, 	0, 	"Distance from wall [m]",  		4.5, 	0.5, 	10.0)
gen.add( 	"x_sp_start",		double_t, 	0, 	"Start setpoint X position [m]",  	0, 	-10.0, 	10.0)
gen.add( 	"y_sp_start",		double_t, 	0, 	"Start setpoint Y position [m]",  	0, 	-10.0, 	10.0)
gen.add( 	"z_sp_start",		double_t, 	0, 	"Start setpoint Z position [m]",  	2.0, 	0, 	10.0)
gen.add( 	"length",		double_t, 	0, 	"Length of the wall to traverse [m]",	2, 	0, 	30.0)
gen.add( 	"height",		double_t, 	0, 	"Height of the wall to traverse [m]",	1, 	0, 	10.0)
gen.add( 	"req_V_overlap_percent",int_t, 		0, "Vertical overlap required for images [%]",	50, 	0, 	100)

gen.add( 	"des_velocity",		double_t, 	0, 	"Absolute Velocity of Circle [m/s]",  	0.5, 	0.1, 	1.5)

exit(gen.generate(PACKAGE, "trajectory_final", "set_trajectory_final"))
